libs/execution/src/lib/types/io-types/filesystem-node-directory.ts

```diff
@@ -0,0 +1,52 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+import { FileSystemNode } from './filesystem-node';
+import { FileSystemFile } from './filesystem-node-file';
+
+export class FileSystemDirectory extends FileSystemNode {
+  private children: FileSystemNode[] = [];
+  constructor(public override name: string) {
+    super(name);
+  }
+
+  override getNodeSize(): number {
+    return this.children.reduce(
+      (totalSize, child) => totalSize + child.getNodeSize(),
+      0,
+    );
+  }
+
+  add(fileSystemNode: FileSystemNode) {
+    this.children.push(fileSystemNode);
+  }
+
+  remove(fileSystemNode: FileSystemNode): FileSystemNode {
+    const index = this.children.indexOf(fileSystemNode);
+    if (index !== -1) {
+      this.children.splice(index, 1);
+    }
+    return this;
+  }
+
+  findFile(fileName: string): FileSystemNode | undefined {
```

Comment:
I'd expect a `FileSystemFile` as a return value here.

Also, you should shift this method to `FileSystemNode` and have the different implementations in its subclasses. That way you can avoid `instanceof` checks.

And also note that this method returns a file with the given name that is found first. To find a specific file at a given path, you need the path as a parameter, so it can be descent recursively along the path.

Reply:
i removed that method since it wasnt used.